src/frameworks/index.ts

```diff
@@ -608,8 +608,8 @@ exports.${functionId} = onRequest(${JSON.stringify(
   }
 }
 
-function codegenDevModeFunctionsDirectory() {
-  const packageJson = {};
+function codegenDevModeFunctionsDirectory(projectPath) {
+  const packageJson = require(`${projectPath}/package.json`)
```

Comment:
Rather than read the package.json, can we change it to `{ engines: { node: NODE_VERSION }}` to address, otherwise we're likely to get extra dependencies and such that the Cloud Functions middleware shim doesn't need.

Reply:
@jamesdaniels 
No because NODE_VERSION is the local value of node.  [see](https://github.com/firebase/firebase-tools/blob/17de2ab9efc85a9ff1fc637fa673ed58e06be874/src/frameworks/index.ts#L111)

For extras, they are deleted [here](https://github.com/firebase/firebase-tools/blob/17de2ab9efc85a9ff1fc637fa673ed58e06be874/src/frameworks/index.ts#L506).
But I would add this:
```suggestion
  delete packageJson.scripts
```

On the other hand, I report an inconsistencies in the code because 

- the version of node must be [>= 16](https://github.com/firebase/firebase-tools/blob/17de2ab9efc85a9ff1fc637fa673ed58e06be874/src/frameworks/index.ts#L261) but only accepts 16 max so we can only use version 16.

-  And package.JSON starts always empty, so the following code is not consistent whether for dev or build
```suggestion
  packageJson.main = "server.js";
      delete packageJson.devDependencies;
      packageJson.dependencies ||= {};
      packageJson.dependencies["firebase-frameworks"] ||= FIREBASE_FRAMEWORKS_VERSION;
      packageJson.dependencies["firebase-functions"] ||= FIREBASE_FUNCTIONS_VERSION;
      packageJson.dependencies["firebase-admin"] ||= FIREBASE_ADMIN_VERSION;
      packageJson.engines ||= {};
      packageJson.engines.node ||= NODE_VERSION;
```

- I also get the message that the firebase-functions version is outdated. 

so I propose that we get Package.json from projectPath in the prepareFramework function


I can code the proposal in this PR

Reply:
**Other proposal:** 

replace `codegenFunctionsDirectory = codegenDevModeFunctionsDirectory;`
by  `codegenFunctionsDirectory = Framework["ecodegenDevModeFunctionsDirectory"] || codegenDevModeFunctionsDirectory` 

So that we are free to return the package.json
that we want depending on the project